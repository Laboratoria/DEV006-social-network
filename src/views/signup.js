function signup(navigateTo) {

  const sectionSignUp = document.createElement('section'); 
  const buttonReturn = document.createElement('button'); 
  const formSignUp = document.createElement('form'); 
  const currentCode=document.createElement("p");
  const userSignUp=document.createElement("p"); 
  const inputUserSignUp = document.createElement('input'); 
  const emailSignUp=document.createElement("p"); 
  const inputEmailSignUp = document.createElement('input'); 
  const passwordSignUp=document.createElement("p"); 
  const inputPasswordSignUp = document.createElement('input'); 
  const confirmPassword=document.createElement("p"); 
  const inputConfirmPasswordSignUp=document.createElement("input"); 
  const buttonSignUp = document.createElement('button'); 

  currentCode.textContent="current code ****"
  inputUserSignUp.placeholder = 'Enter user'; 
  inputEmailSignUp.placeholder = 'Enter email'; 
  inputPasswordSignUp.placeholder="Enter password"; 
  inputConfirmPasswordSignUp.placeholder="confirm password";
  userSignUp.textContent="User name"; 
  emailSignUp.textContent="E-mail"; 
  passwordSignUp.textContent="Password"; 
  confirmPassword.textContent="Confirm password"; 
  buttonSignUp.textContent = 'SignUp'; 
  buttonReturn.textContent = 'Back'; 
  buttonReturn.classList.add("buttonReturn");

  formSignUp.classList.add("container-formSignUp");
  currentCode.classList.add("parrafo-signUp")
  userSignUp.classList.add("parrafo-signUp");
  emailSignUp.classList.add("parrafo-signUp");
  passwordSignUp.classList.add("parrafo-signUp");
  confirmPassword.classList.add("parrafo-signUp");
  inputUserSignUp.classList.add("input-userSignUp");
  inputEmailSignUp.classList.add("input-userSignUp"); 
  inputPasswordSignUp.classList.add("input-userSignUp");
  inputConfirmPasswordSignUp.classList.add("input-userSignUp");
  buttonSignUp.classList.add("btn-windowSignUp");
  


  buttonReturn.addEventListener('click', () => { 

    navigateTo('/'); 

  }); 

 
  formSignUp.append(currentCode,userSignUp,inputUserSignUp,emailSignUp, inputEmailSignUp,passwordSignUp,inputPasswordSignUp,confirmPassword,inputConfirmPasswordSignUp,buttonSignUp); 
  sectionSignUp.append(buttonReturn,formSignUp);
  
  
  return sectionSignUp; 

} 

 
 

export default signup; 

 
  /*const section = document.createElement('section');

  const title = document.createElement('h2');
  const buttonReturn = document.createElement('button');
  // const form = document.createElement('form');
  const inputEmail = document.createElement('input');
  const inputPass = document.createElement('input');
  const buttonLogin = document.createElement('button');

  const container = document.createElement('div');
  container.className = 'container-app';

  inputEmail.placeholder = 'Write email';
  inputEmail.className = 'input-email';
  inputPass.placeholder = 'pass';
  inputPass.classList.add('inputPassword');

  title.textContent = 'Login';
  title.className = 'title';
  buttonLogin.textContent = 'go';
  buttonLogin.classList.add('buttonLogin');

  buttonReturn.textContent = 'Return to home';
  buttonReturn.classList.add('buttonReturn');

  buttonReturn.addEventListener('click', () => {
    navigateTo('/');
  });


  container.append(title, inputEmail, inputPass, buttonLogin, buttonReturn);
  section.append(container);


  return section;
}

export default login;
// function signUp (navigateTo) {
//   const form2 = document.createElement ('form');
//   const userName = document.createElement ('p');
//   const inputUserName = document.createElement ('input');
//   const email = document.createElement ('p');
//   const inputEmail = document.createElement ('input');
//   const password = document.createElement ('p');
//   const inputPassword = document.createElement ('input');
//   const confirmPassword = document.createElement ('p');
//   const inputConfirmPassword = document.createElement ('input');
//   const btnSignUp = document.createElement ('button');
//   const btnReturn = document.createElement ('button');
//   const section = document.createElement ('section');

//   inputUserName.placeholder = 'Write username';
//   inputEmail.placeholder = 'Write email';
//   inputPassword.placeholder = 'Write password';
//   inputConfirmPassword.placeholder = 'Confirm password';

//   btnSignUp.textContent = 'Sign Up';

//   btnReturn.textContent = 'Return';
//   btnReturn.addEventListener('click',() => {
//     navigateTo('/');
//   })

//   form2.append (inputUserName, inputEmail, inputPassword, inputConfirmPassword, btnSignUp );
//   section.append (form2, btnReturn);

//   return section;
// }

// export default signUp;*/
