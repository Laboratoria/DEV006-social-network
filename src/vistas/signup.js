import { registerUser, signInWithGoogle } from '../lib/auth.js';

function signup(navigateTo) {
  // creaci칩n de elementos//
  const section = document.createElement('section');
  const sectionForm = document.createElement('section');
  const sectionHeader = document.createElement('section');
  const UserSignup = document.createElement('h4');
  const emailSignup = document.createElement('h4');
  const passwordSignup = document.createElement('h4');
  const form = document.createElement('form');
  const inputUser = document.createElement('input');
  const inputEmail = document.createElement('input');
  const inputPassword = document.createElement('input');
  const passwordBox = document.createElement('div');
  const buttonEnterSignup = document.createElement('button');
  const buttonReturnSignup = document.createElement('img');
  const showPassword = document.createElement('img');
  const logo = document.createElement('img');
  const header = document.createElement('div');
  const errorPassword = document.createElement('span');
  // const errorEmail = document.createElement('span');
  const buttonGoogleSignup = document.createElement('button');

  inputEmail.placeholder = 'example@gmail.com';
  inputPassword.placeholder = 'Ingresa minimo 6 caracteres';
  inputUser.placeholder = 'Ingresa nombre usuaria';
  // agregar atributos//
  logo.setAttribute('src', 'images/logo.png');
  errorPassword.setAttribute('id', 'errorPassword');
  // errorEmail.setAttribute('id', 'errorEmail');//
  buttonReturnSignup.setAttribute('src', 'images/arrow.png');
  inputEmail.setAttribute('id', 'inputEmail');
  inputPassword.setAttribute('id', 'inputPassword');
  inputUser.setAttribute('id', 'inputUser');
  inputPassword.setAttribute('type', 'password');
  inputPassword.setAttribute('src', 'images/ojoOculto.png');
  showPassword.setAttribute('src', 'images/ojoOculto.png');
  passwordBox.setAttribute('id', 'passwordBox');
  buttonGoogleSignup.textContent = 'Registrarse con Google';
  emailSignup.textContent = 'Correo electr칩nico';
  passwordSignup.textContent = 'Contrase침a';
  UserSignup.textContent = 'Nombre de Usuario';
  buttonEnterSignup.textContent = 'Registrarte';

  // Registro con Google
  buttonGoogleSignup.addEventListener('click', (e) => {
    e.preventDefault();
    signInWithGoogle();
  });
  // Registro de usuario//
  buttonEnterSignup.addEventListener('click', (e) => {
    e.preventDefault();
    const email = inputEmail.value;
    const password = inputPassword.value;
    const user = inputUser.value;
    // errorEmail.textContent = '';
    errorPassword.textContent = '';
    registerUser(email, password, user)
      .then((userResult) => {
        console.log('signup', userResult);
        inputEmail.value = '';
        inputPassword.value = '';
        inputUser.value = '';
        navigateTo('/wall');
      })

      .catch((error) => {
        console.log(error);
        // errorEmail.innerHTML = error;
        errorPassword.innerHTML = error;
        // const codeError = error.code;
        // errorMessages(codeError, errorEmail, errorPassword);
      });
  });

  // ruteado para button de retornar
  buttonReturnSignup.addEventListener('click', () => {
    navigateTo('/');
  });
  // Ocultar y mostrar contrase침a//
  showPassword.addEventListener('click', (e) => {
    e.preventDefault();
    if (inputPassword.type === 'password') {
      inputPassword.type = 'text';
      showPassword.src = 'images/ojonoOculto.png';
    } else {
      inputPassword.type = 'password';
      showPassword.src = 'images/ojoOculto.png';
    }
  });

  // // agregar clases//

  header.classList.add('header');
  logo.classList.add('logo');
  sectionForm.classList.add('sectionFormSignup');
  sectionHeader.classList.add('sectionHeaderSignup');
  form.classList.add('formSignup');
  buttonReturnSignup.classList.add('buttonReturnSignup');
  UserSignup.classList.add('subTitlesSignup');
  emailSignup.classList.add('subTitlesSignup');
  passwordSignup.classList.add('subTitlesSignup');
  buttonEnterSignup.classList.add('buttonEnterSignup');
  errorPassword.classList.add('errors');
  // errorEmail.classList.add('errors');
  showPassword.classList.add('showPassword');
  buttonGoogleSignup.classList.add('buttonGoogleSignup');

  // agrupar secciones//
  passwordBox.append(inputPassword, showPassword);
  form.append(
    UserSignup,
    inputUser,
    emailSignup,
    inputEmail,
    // errorEmail,
    passwordSignup,
    passwordBox,
    errorPassword,

  );
  sectionForm.append(form, buttonEnterSignup, buttonGoogleSignup);
  sectionHeader.append(header, logo);
  section.append(sectionHeader, buttonReturnSignup, sectionForm);

  return section;
}
export default signup;
