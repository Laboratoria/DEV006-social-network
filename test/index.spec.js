/**
 * @jest-environment jsdom
 */
// importamos la funcion que vamos a testear
import signup from '../src/vistas/signup.js';

//Validacion de funcion Snapshot de vista
describe('signup', () => {
  it('snapshot', () => {
    const vista = signup(() => {});
    const container = document.createElement('div');
    container.append(vista);
    expect(container).toMatchSnapshot();
  });
  //Validacion de boton de retorno a la pagina home 
  it('back', () => {
    const navigateTo = jest.fn();//(es una funcion mock)
    const vista = signup(navigateTo);
    const container = document.createElement('div');
    container.append(vista);
    const botonBack = container.querySelector('.buttonReturnSignup');
    botonBack.click();
    expect(navigateTo).toHaveBeenCalled();
  });
});
