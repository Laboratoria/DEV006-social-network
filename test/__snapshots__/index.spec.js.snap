// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signup snapshot 1`] = `
<div>
  <section>
    <section
      class="sectionHeaderSignup"
    >
      <div
        class="header"
      />
      <img
        class="logo"
        src="images/logo.png"
      />
    </section>
    <img
      class="buttonReturnSignup"
      src="images/arrow.png"
    />
    <section
      class="sectionFormSignup"
    />
    <form
      class="formSignup"
    >
      <h4
        class="subTitlesSignup"
      >
        Nombre de Usuario
      </h4>
      <input />
      <h4
        class="subTitlesSignup"
      >
        Correo electrónico
      </h4>
      <input
        id="inputEmail"
        placeholder="example@gmail.com"
      />
      <h4
        class="subTitlesSignup"
      >
        Contraseña
      </h4>
      <input
        id="inputPassword"
        placeholder="***********"
        src="images/ojoOculto.png"
        type="password"
      />
      <button
        class="buttonEnterSignup"
      >
        Registrarte
      </button>
      <h5
        class="question"
      >
        Esta aplicación esta pensada para personas que se identifican como mujeres. Por favor responde: ¿Te identificas como mujer?  
      </h5>
      <input
        id="Yes"
        name="checkbox"
        type="radio"
      />
      <input
        id="No"
        name="checkbox"
        type="radio"
      />
      <h5
        id="labelYes"
      >
        Si
      </h5>
      <h5
        id="labelNo"
      >
        No
      </h5>
      <img
        id="showPasswordLogin"
        src="images/ojoOculto.png"
      />
    </form>
  </section>
</div>
`;
